submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/fire/FireFilesService.kt

```diff
@@ -56,15 +64,23 @@ fun FireFileProcessingConfig.processFile(sub: ExtSubmission, file: ExtFile): Ext
 fun FireFileProcessingConfig.processNfsFile(relPath: String, nfsFile: NfsFile): ExtFile {
     logger.info { "$accNo $owner Persisting file ${nfsFile.fileName} with size ${nfsFile.file.size()} on FIRE" }
 
-    return when {
-        nfsFile.file.isDirectory -> persistFireDirectory(nfsFile)
-        else -> persistFireFile(accNo, relPath, nfsFile)
-    }
+    return if (nfsFile.file.isDirectory) persistFireDirectory(nfsFile) else persistFireFile(accNo, relPath, nfsFile)
 }
 
-private fun persistFireDirectory(nfsFile: NfsFile): FireDirectory {
+private fun FireFileProcessingConfig.persistFireDirectory(nfsFile: NfsFile): FireDirectory {
```

Comment:
here I think we need to do something, we are adding  this logic to create file and directory structure in this class which is supose to be in charge of business logic. I think we need to extract this part somewhere else.  Lets meet and discuss about this one as we have similar requirements in others parts.